presto-main/src/main/java/com/facebook/presto/sql/planner/PlannerUtils.java

```diff
@@ -87,4 +123,155 @@ public static VariableReferenceExpression toVariableReference(Expression express
+    {
+        PlanNode newSource = clonePlanNode(filterNode.getSource(), session, metadata, planNodeIdAllocator, variablesToKeep, varMap);
+
+        return new FilterNode(
+                filterNode.getSourceLocation(),
+                idAllocator.getNextId(),
+                newSource,
+                filterNode.getPredicate());
+    }
+
+    private static PlanNode cloneProjectNode(ProjectNode projectNode, Session session, Metadata metadata, PlanNodeIdAllocator planNodeIdAllocator, List<VariableReferenceExpression> fieldsToKeep, Map<VariableReferenceExpression, VariableReferenceExpression> varMap, PlanNodeIdAllocator idAllocator)
+    {
+        PlanNode newSource = clonePlanNode(projectNode.getSource(), session, metadata, planNodeIdAllocator, fieldsToKeep, varMap);
+
+        return new ProjectNode(
+                idAllocator.getNextId(),
+                newSource,
+                projectNode.getAssignments());
+    }
+
+    private static TableScanNode cloneTableScan(TableScanNode scanNode, Session session, Metadata metadata, PlanNodeIdAllocator planNodeIdAllocator, List<VariableReferenceExpression> fieldsToKeep, Map<VariableReferenceExpression, VariableReferenceExpression> varMap)
+    {
+        Map<VariableReferenceExpression, ColumnHandle> assignments = scanNode.getAssignments();
+
+        TableLayout scanLayout = metadata.getLayout(session, scanNode.getTable());
+
+        return new TableScanNode(
+                scanNode.getSourceLocation(),
+                planNodeIdAllocator.getNextId(),
+                scanLayout.getNewTableHandle(),
+                scanNode.getOutputVariables(),
+                scanNode.getAssignments(),
+                scanNode.getTableConstraints(),
+                scanNode.getCurrentConstraint(),
+                scanNode.getEnforcedConstraint());
+    }
+
+    public static PlanNode clonePlanNode(PlanNode planNode, Session session, Metadata metadata, PlanNodeIdAllocator planNodeIdAllocator, List<VariableReferenceExpression> fieldsToKeep, Map<VariableReferenceExpression, VariableReferenceExpression> varMap)
+    {
+        if (planNode instanceof TableScanNode) {
+            TableScanNode scanNode = (TableScanNode) planNode;
+            return cloneTableScan(scanNode, session, metadata, planNodeIdAllocator, fieldsToKeep, varMap);
+        }
+        else if (planNode instanceof FilterNode) {
+            return cloneFilterNode((FilterNode) planNode, session, metadata, planNodeIdAllocator, fieldsToKeep, varMap, planNodeIdAllocator);
+        }
+        else if (planNode instanceof ProjectNode) {
+            return cloneProjectNode((ProjectNode) planNode, session, metadata, planNodeIdAllocator, fieldsToKeep, varMap, planNodeIdAllocator);
+        }
+
+        checkState(false, "Currently cannot clone: " + planNode.getClass().getName() + " nodes.");
+        return null;
+    }
+
+    public static String getPlanString(PlanNode planNode, Session session, TypeProvider types, Metadata metadata)
```

Comment:
Not used, and its just a wrapper on PlanPrinter API. Can we remove it?

Reply:
It's a util function useful for debugging

Reply:
Lets remove it if we are not using it.

Reply:
No this is a util function that useful for debugging.